foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/mappings/structures/ObjectRelationalDataTypeDescriptor.java

```diff
@@ -336,7 +337,13 @@ public Struct buildStructureFromRow(AbstractRecord row, AbstractSession session,
             Object[] fields = new Object[getOrderedFields().size()];
             for (int index = 0; index < getOrderedFields().size(); index++) {
                 DatabaseField field = (DatabaseField)getOrderedFields().elementAt(index);
-                fields[index] = row.get(field);
+                if (row.getField(field) != null && "java.sql.Blob".equals(row.getField(field).getTypeName())) {
```

Comment:
check either against `ClassConstants.BLOB`/`DBField.getType()` or  `JDBCType.BLOB`/`DBField.getSqlType` is better thing to do

Reply:
I agree, but `field` properties behind `field/row.getField(field) .getType()` or `field/row.getField(field) .getSqlType()` are not available at this moment. Only `typeName` is there.

Reply:
Ok then. What about `Clob`? The title mentions it but I don't see it explicitly handled

Reply:
I'm still not sure if conversion like this should be there. Maybe more correct way should be call something like
`fields[index] = session.getDatasourcePlatform().convertObject(row.get(field), java.sql.Blob.class)...` or something more generic
instead of 
`for (int index = 0; index < getOrderedFields().size(); index++) {....}`?

What I see little problematic in this way is that "live" instance of `java.sql.Connection` for `java.sql.B(C)lob` creation is needed. Is it correct or possible use `java.sql.Connection` in `org.eclipse.persistence.platform.database.*Platform` classes?
But I see `org.eclipse.persistence.platform.database.*Platform` classes (`org.eclipse.persistence.platform.database.oracle.Oracle12Platform` in this particular case) as a better location.

Reply:
there is a call to platform specific method already a line bellow the code you've added - can the logic be moved there, into the `session.getPlatform().createStruct(getStructureName(), fields, session, connection);` method? The _bad_ field is there in the `fields` variable..